### PR TITLE
[CRIMAPP-1044] Fix outgoings summary title

### PR DESCRIPTION
**Pull request by mseedat-moj**
## Description of change
Fix outgoings summary title

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-1044
Revision as a result of this PR: https://dsdmoj.atlassian.net/browse/CRIMAPP-1082 (not yet implemented)

## Notes for reviewer
Should fix any other similar bugs
### Before changes:
Screenshot 2024-06-12 at 15 16 59-20240612-141705

### After changes:

Screenshot 2024-06-13 at 10 52 25
Screenshot 2024-06-13 at 10 54 33

## How to manually test the feature
Complete outgoings

### CODE REVIEW

File: app/presenters/summary/sections/base_section.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def title
         I18n.t(
           name,
           scope: 'summary.sections',
-          subject: I18n.t('summary.dictionary.subject', subject_type:),
+          subject: I18n.t("summary.dictionary.subjects.#{subject_type}"),
```
Comment (timpeat) [inline app/presenters/summary/sections/base_section.rb:32]: @mseedat-moj is this change necessary? This is handled here: 
https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/blob/main/config/locales/en.rb

and is used elsewhere I think.

Comment (timpeat) [reply app/presenters/summary/sections/base_section.rb:32]: @mseedat-moj forgive me, I see this is for the summary dictionary.

File: config/locales/en/summary.yml [hunk 2]
```diff
@@ -66,13 +66,13 @@ en:
...
       income_payments_details: Payments your client gets
       income_benefits_details: Benefits your client gets
-      other_income_details: 
+      other_income_details:
         one: "Other sources of income"
         other: "Other sources of income: client and partner"
       housing_payments: Housing Payments
-      outgoings_payments_details: 
-        one: Payments %{subject} makes
-        other: Payments %{subject} make
```
Comment (timpeat) [inline config/locales/en/summary.yml:75]: This will incorrectly produce the title "Payments your partner makes" I think. Given this has a different subject (__your__ client, __their__ partner) to other titles (partner, client). I'd be inclined to overwrite the title method in the child class with something that uses the same subject as the forms. I.e.

def title
  I18n.t(
          name,
          scope: 'summary.sections',
          subject: I18n.t("dictionary.subjects.#{subject_type}")
        ) 
        end

Comment (timpeat) [reply config/locales/en/summary.yml:75]: Also. I don't think think this text will every be plural.

**Review by ripan — APPROVED** [review]

File: app/presenters/summary/sections/base_section.rb (hunk 1 above)
Comment (arthurashman) [inline app/presenters/summary/sections/base_section.rb:32]: I'm not sure subject_type is working on the outgoings page. The page title knows that it is the client and partner, but the section title doesn't
image

Comment (arthurashman) [reply app/presenters/summary/sections/base_section.rb:32]: I think we want ownership too, not subject looking at the dictionary

File: config/locales/en/summary.yml (hunk 2 above)
Comment (arthurashman) [reply config/locales/en/summary.yml:75]: Pluralisation is to avoid:
`Payments your client and their partner **make**`

Comment (timpeat) [reply config/locales/en/summary.yml:75]: Yes @arthurashman but in this context it will only ever be client OR partner. No?

File: config/locales/en/summary.yml [hunk 3]
```diff
@@ -66,13 +66,13 @@ en:
...
+      outgoings_payments_details:
+        one: Payments your %{subject} makes
+        other: Payments your %{subject} make
```
Comment (arthurashman) [inline config/locales/en/summary.yml:75]: If we switch to ownership from the dictionary, you need to cut these 'your's

File: config/locales/en/summary.yml (hunk 2 above)
Comment (arthurashman) [reply config/locales/en/summary.yml:75]: Nope, for outgoings the payments page is either `Client` or `Client and Partner`
